Backend/AlgorithmsManager.py

```diff
@@ -339,28 +342,50 @@ def get_economic_data_by_state(self, state):
         all_cities_data = []
 
         for entry in resrows:
-            city_data = {}
-            city_data["woe_id"] = entry[1]; city_data["city"] = entry[2]
-            city_data["state"] = entry[3]; city_data["lat"] = entry[4]
-            city_data["long"] = entry[5]; city_data["population"] = entry[7]
-            city_data["density"] = entry[8]; city_data["ranking"] = entry[9]
-            city_data["age_median"] = entry[10]; city_data["male"] = entry[11]
-            city_data["female"] = entry[12]; city_data["married"] = entry[13]
-            city_data["family_size"] = entry[14]
-            city_data["income_household_median"] = entry[15]
-            city_data["income_six_figure_percent"] = entry[16]; city_data["home_ownership"] = entry[17]
-            city_data["home_value"] = entry[18]; city_data["rent_median"] = entry[19]
-            city_data["education_college_or_above"] = entry[20]
-            city_data["labor_force_participation"] = entry[21]
-            city_data["unemployment_rate"] = entry[22]
-            city_data["race_white"] = entry[23]; city_data["race_black"] = entry[24]
-            city_data["race_asian"] = entry[25]; city_data["race_native"] = entry[26]
-            city_data["race_pacific"] = entry[27]; city_data["race_other"] = entry[28]
-            city_data["race_multiple"] = entry[29]
-            all_cities_data.append(city_data)
+            city = self.parse_city_data(entry)
+            all_cities_data.append(city)
 
         return all_cities_data
 
+    # retrieves economic data for a city or woeid
+    #   arg: city (full name), woeid (either str or int)
+    #   return type: dict
+    def get_economic_data_by_city(self, city, woeid):
+        if city is not None:
+            queryres = self.database.query("SELECT * FROM city_social_data WHERE city = (%s);", city)
+
+        elif woeid is not None:
+            queryres = self.database.query("SELECT * FROM city_social_data WHERE woe_id = (%s);", woeid)
```

Comment:
the function's doc comment says that woeid can be either a string value or an int. Does this line work equally well in either case? I'd thought that that kind of 'insert something into a string' code was sensitive to that kind of data type distinction. In particular, I thought the insertion code "%s" was specific to string values.
Would it work to do a check, inside this woeid if block, to see whether the woeid was an int value, and if it was then it could be converted to a string before this database query line?

Reply:
I tested it and it works. 

`algo.get_economic_data_by_city(None, "2459115")` and `algo.get_economic_data_by_city(None, 2459115)` work equally well

Reply:
Oh, brilliant, thanks!